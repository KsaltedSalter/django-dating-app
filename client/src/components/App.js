import React, { useState, useEffect } from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { isLoggedIn } from "../api/authToken";

import Home from "./common/Home";
import Navbar from "./common/Navbar";
import ProfileList from "./personas/ProfileList";
import UserProfile from "./personas/UserProfile";
import Login from "./reglog/Login";
import Registration from "./reglog/Registration";
// import SecureRoute from "./common/SecureRoute";

function App() {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  useEffect(() => {
    if (isLoggedIn()) {
      setIsAuthenticated(true);
    }
  }, []);

  return (
    <BrowserRouter>
      <Navbar isAuthenticated={isAuthenticated} />
      <Routes>
        <Route exact path="/" element={<Home />} />
        <Route path="/myprofile/:id" element={<UserProfile />} />
        <Route path="/search" element={<ProfileList />} />
        <Route path="/register" element={<Registration />} />
        <Route path="/login" element={<Login />} />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
