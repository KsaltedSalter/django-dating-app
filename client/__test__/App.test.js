import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../src/components/App";

describe("<App />", () => {
  it("matches snapshot on initial render", () => {
    const component = render(<App />);
    expect(component.baseElement).toMatchSnapshot();
  });
});
