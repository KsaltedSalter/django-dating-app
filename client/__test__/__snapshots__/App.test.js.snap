// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot on initial render 1`] = `
<body>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container-fluid"
      >
        <span
          class="navbar-brand mb-0 h1"
        >
          <img
            alt="ani-mate logo"
            class="d-inline-block align-text-top logo-img"
            height="35"
            src=""
            width="40"
          />
          Ani-mate
        </span>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler navbar-dark"
          data-bs-target="#navbarNavAltMarkup"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon navbar-dark"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            class="navbar-nav ms-auto"
          >
            <a
              class="nav-link"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              href="/register"
            >
              Register
            </a>
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="hero-container"
    >
      <img
        alt="couple confessing love"
        class="hero"
        src=""
      />
      <p
        class="hero-text text-center"
      >
        Welcome to Ani-mate! 
        <br />
        An exclusive dating site for the 2D lonely hearts looking for their one true love. 
        <br />
        Tired of not finding "the one" in your universe? 
        <br />
        Too busy fighting those villians to meet someone new?
        <br />
        Then look no further, and register above to start your journey. 
        <br />
         
        Believe it! 
        <br />
        がんばって!
      </p>
    </div>
  </div>
</body>
`;
